packages/liveblocks-react-tiptap/src/mentions/MentionsList.tsx

```diff
@@ -0,0 +1,166 @@
+        {isLoading ? null : name}
+      </span>
+    );
+  }
+);
+
+
+export const SUGGESTIONS_COLLISION_PADDING = 10;
+
+export type MentionsListProps = {
+  query: string,
+  command: (otps: { id: string, notificationId: string }) => void,
+  clientRect: () => DOMRect,
+  hide: boolean
+};
+export type MentionsListHandle = {
+  onKeyDown: ({ event }: { event: KeyboardEvent }) => boolean,
+};
+
+export const MentionsList = forwardRef<MentionsListHandle, MentionsListProps>((props, ref) => {
+  const [selectedIndex, setSelectedIndex] = useState(0);
+  const suggestions = useMentionSuggestions(props.query);
+  const {
+    refs: { setReference, setFloating },
+    strategy,
+    x,
+    y,
+  } = useFloating({
+    strategy: "fixed",
+    placement: "top-start",
+    middleware: [
+      flip({ padding: SUGGESTIONS_COLLISION_PADDING, crossAxis: false }),
+      offset(10),
+      hide({ padding: SUGGESTIONS_COLLISION_PADDING }),
+      shift({ padding: SUGGESTIONS_COLLISION_PADDING, limiter: limitShift() }),
+      size({ padding: SUGGESTIONS_COLLISION_PADDING }),
+    ],
+    whileElementsMounted: (...args) => {
+      return autoUpdate(...args, {
+        animationFrame: true,
+      });
+    },
+  });
+
+  useLayoutEffect(() => {
+    setReference({
+      getBoundingClientRect: props.clientRect,
+    });
+  }, [setReference, props.clientRect]);
+
+  const selectItem = (index: number) => {
+    const item = (suggestions ?? [])[index];
+    if (item) {
+      props.command({ id: item, notificationId: createInboxNotificationId() });
+    }
+  }
+
+  const upHandler = () => {
+    setSelectedIndex((selectedIndex + (suggestions?.length ?? 0) - 1) % (suggestions?.length ?? 0))
+  }
+
+  const downHandler = () => {
+    setSelectedIndex((selectedIndex + 1) % (suggestions?.length ?? 0))
+  }
+
+  const enterHandler = () => {
+    selectItem(selectedIndex)
+  }
+
+  useEffect(() => setSelectedIndex(0), [suggestions])
+
+  useImperativeHandle(ref, () => ({
+    onKeyDown: ({ event }: { event: KeyboardEvent }) => {
+      if (event.key === "ArrowUp") {
+        upHandler()
+        return true
+      }
+
+      if (event.key === "ArrowDown") {
+        downHandler()
+        return true
+      }
+
+      if (event.key === "Enter") {
+        enterHandler()
+        return true
+      }
+
+      return false
+    },
+  }))
+
+  if (suggestions !== undefined && suggestions.length === 0) {
+    return null
+  }
+
+  return (
+    <div className="lb-root lb-portal lb-elevation lb-tiptap-suggestions lb-tiptap-mention-suggestions" ref={setFloating}
+      style={{
+        position: strategy,
+        top: 0,
+        left: 0,
+        transform: `translate3d(${Math.round(x)}px, ${Math.round(y)}px, 0)`,
+        minWidth: "max-content",
+        display: props.hide ? "none" : "block"
+      }}>
+      <div className="lb-tiptap-suggestions-list lb-tiptap-mention-suggestions-list">
+        {suggestions === undefined ? <div className="item">Loading...</div> :
```

Comment:
We need to add the real loading placeholder here.

Reply:
In the Comments version, we don't display the dropdown at all while it loads so this check could be added to line 129 to render `null`